js/app.js

```
@@ -198,36 +198,68 @@ function tagIsSelected(blogTag) {
+        elementSnippet = document.createElement(snippetType.tag);
+        if (!elementSnippet) return;
         const snippetContent = firstOrDefaultValue(data[key], snippetType.property) || "";
-        if (snippetType.format === "text") elementSnippet.textContent = snippetContent;
-        else elementSnippet.innerHTML = snippetContent;
-        if (typeCode === "aside_note") {
-            const asideStyle = firstOrDefaultValue(data[key], "note_style");
-            if (asideStyle) elementSnippet.classList.add("aside-" + asideStyle.codename);
-        }
-        objSnippet.parentNode.replaceChild(elementSnippet, objSnippet);
+        setElementContent(snippetType, snippetContent);
+        setElementClass(snippetType);
+        setElementOptions(data, key);
+        loadElementSnippet();
     }
 }
+function setElementContent(snippetType, snippetContent) {
+    if (snippetType.format === "text") elementSnippet.textContent = snippetContent;
+    else elementSnippet.innerHTML = snippetContent;
+}
+function setElementClass(snippetType) {
+    if (snippetType.classname) elementSnippet.classList.add(snippetType.classname);
+}
+function setElementOptions(data, key) {
+    tryProcessAsideNote(data, key);
+    tryProcessQuotation(data, key);
+}
+function tryProcessAsideNote(data, key) {
+    try {
+        const asideStyle = firstOrDefaultValue(data[key], "note_style");
+        if (asideStyle) elementSnippet.classList.add("aside-" + asideStyle.codename);
+    } catch { /* no need */ }
+}
+function tryProcessQuotation(data, key) {
+    try {
+        const quoteReference = firstOrDefaultValue(data[key], "reference");
+        if (quoteReference) {
+            var elementReference = document.createElement('footer');
+            elementReference.innerHTML = quoteReference;
+            elementReference.classList.add("quote-reference");
+            elementSnippet.appendChild(elementReference);
+        }
+    } catch { /* no need */ }
```

What are these empty catch blocks about? Definite code smell.